strawberry_django_plus/utils/resolvers.py

```diff
@@ -413,25 +418,29 @@ def resolve_model_node(source, node_id, *, info: Optional[Info] = None, required
     return ret
 
 
-def resolve_model_id(root: Model) -> AwaitableOrValue[str]:
+def resolve_model_id(source: Union[Type[Node], Type[_M]], root: Model) -> AwaitableOrValue[str]:
     """Resolve the model id, ensuring it is retrieved in a sync context.
 
     Args:
+        source:
+            The source model or the model type that implements the `Node` interface
         root:
             The source model object.
 
     Returns:
         The resolved object id
 
     """
+    id_attr = getattr(source, "id_attr", "pk")
     assert isinstance(root, Model)
-    pk = root.__class__._meta.pk
-    assert pk
+    if id_attr == "pk":
+        id_attr = root.__class__._meta.pk.attname
```

Comment:
You will probably need to do this to fix pyright issue (the typing says `pk` can be `None`)

```suggestion
        pk = root.__class__._meta.pk
        assert pk
        id_attr = pk.attname
```

Reply:
thanks, will include this